### ISSUE TITLE
Separate entire changed lines more clearly

### ISSUE THREAD

**Issue by joelostblom** [trevorcampbell/website_diff#26]
This source diff:

![image](https://github.com/user-attachments/assets/affe8d01-8318-4b40-9be5-1ce8e1e92a20)

is represented as this rendered diff:

![image](https://github.com/user-attachments/assets/a8c93580-cb9b-4e63-a385-2407ec5f4251)

I think the source diff is clearer to read in this case, given that the change has resulted in a completely new sentence rather than just word-level modifications to the same sentence. Not sure how easy this is to split out on the render level, but it could be clearer to try to do what the source diff does here and mark one entire sentence as deleted and then one entire sentence as added instead of doing it word by work.

Live here for a bit: https://joelostblom.github.io/viz-oer/diff29/index.html

**Comment by trevorcampbell** [comment]
Yeah, it's a known behaviour. I don't mind it for now, since the big win of `website_diff` is to catch diffs that aren't obvious from the source files (in your example, it's already obvious there). But it would be nice to collapse these tiny little piecemeal insertion/deletion elements.

For posterity, the source of the diff looks like:

![image](https://github.com/user-attachments/assets/171d792e-58b0-4e6a-9ecf-ecdb4580f1d9)

**Comment by trevorcampbell** [comment]
@briank-git does the diff engine have some flags that let us tune that? If not, we can always just check for sequences of `<ins>` and `<del>` elements that just contain text (no internal elements) and merge those, as they're commutative.

**Comment by joelostblom** [comment]
Yes, agree this is a minor convenience and that the big win is to notify me of things I wouldn't see otherwise :+1:

**Comment by briank-git** [comment]
@trevorcampbell No the diff engine doesn't have those flags, its pretty bare bones.

**Comment by trevorcampbell** [comment]
Understood. Yeah in theory, I think that we can actually merge any sibling `<ins>` elements and `<del>` elements in the tree (even if they contain other stuff below), so maybe that's an easy "post-processing" step that we can do at some point. Not urgent though
